flask_security/registerable.py

```diff
@@ -40,4 +39,7 @@ def register_user(**kwargs):
                             'welcome', user=user,
                             confirmation_link=confirmation_link)
 
+        if _security.confirmable:
```

Comment:
@kantorii can you please clarify why do we need to move the `do_flash` call to the separate block?

Reply:
Yes. If there is an exception within user_registered.send(), do_flash() will not be executed.
I believe it should be that way, because, if I remember correctly, the flash says the message was sent.